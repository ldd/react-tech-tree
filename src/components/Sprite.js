import React from "react";
import "./Sprite.css";

export const Sprite = ({
  className = "",
  name,
  styleName = () => {},
  ...otherProps
}) => {
  const spriteStyle = {
    ...styleName(name)
  };
  return (
    <div
      className={`${className} Sprite`}
      style={spriteStyle}
      {...otherProps}
    />
  );
};
