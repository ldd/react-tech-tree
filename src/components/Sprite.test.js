import React from "react";
import renderer from "react-test-renderer";
import { Sprite } from "./Sprite";

describe("React Component [Sprite]", () => {
  it("can create empty Sprite", () => {
    const component = renderer.create(<Sprite />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("can create Sprite with extra props", () => {
    const className = "someClass";
    const someProp = "@value";
    const component = renderer.create(
      <Sprite className={className} someProp={someProp} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
