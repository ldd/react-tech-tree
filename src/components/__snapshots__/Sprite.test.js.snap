// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component [Sprite] can create Sprite with extra props 1`] = `
<div
  className="someClass Sprite"
  someProp="@value"
  style={Object {}}
/>
`;

exports[`React Component [Sprite] can create empty Sprite 1`] = `
<div
  className=" Sprite"
  style={Object {}}
/>
`;
